apps/web/app/[orgShortCode]/settings/SettingsSidebar.tsx

```diff
@@ -1,11 +1,211 @@
 'use client';
-
-import { Flex } from '@radix-ui/themes';
+import { useEffect, useState } from 'react';
+import { useParams, usePathname } from 'next/navigation';
+import { Flex, Text } from '@radix-ui/themes';
+import Link from 'next/link';
+import {
+  CircleUser,
+  Mail,
+  Lock,
+  Building2,
+  CreditCard,
+  AtSign,
+  Globe,
+  Users,
+  UserRoundPlus,
+  HeartHandshake
+} from 'lucide-react';
 
 export default function SettingsSidebar() {
+  const pathname = usePathname();
+  const [currentUrl, setCurrentUrl] = useState('');
+  const { orgShortCode } = useParams();
+
+  useEffect(() => {
+    const segments = pathname.split('/').filter(Boolean);
+    if (segments.length > 0) {
+      let endpoint = segments[segments.length - 1];
+      if (endpoint === 'addresses') {
+        endpoint = `${segments[segments.length - 2]}/${segments[segments.length - 1]}`;
+      }
+      setCurrentUrl(endpoint || '');
+    }
+  }, [pathname]);
```

Comment:
we don't really need this, just use the path url as key

Reply:
ahh, I overcooked didn't thought of this :)